cpp/oneapi/dal/algo/logistic_regression/backend/gpu/train_kernel_dense_batch_dpc.cpp

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/profiler.hpp"
+
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/backend/dispatcher.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/lapack.hpp"
+#include "oneapi/dal/backend/primitives/utils.hpp"
+
+#include "oneapi/dal/table/row_accessor.hpp"
+
+#include "oneapi/dal/algo/logistic_regression/common.hpp"
+#include "oneapi/dal/algo/logistic_regression/train_types.hpp"
+#include "oneapi/dal/algo/logistic_regression/backend/model_impl.hpp"
+#include "oneapi/dal/algo/logistic_regression/backend/gpu/train_kernel.hpp"
+#include "oneapi/dal/backend/primitives/objective_function.hpp"
+#include "oneapi/dal/backend/primitives/optimizers.hpp"
+#include "oneapi/dal/algo/logistic_regression/backend/optimizer_impl.hpp"
+
+namespace oneapi::dal::logistic_regression::backend {
+
+using dal::backend::context_gpu;
+
+namespace be = dal::backend;
+namespace pr = be::primitives;
+
+template <typename Float, typename Task>
+static train_result<Task> call_dal_kernel(const context_gpu& ctx,
+                                          const detail::descriptor_base<Task>& desc,
+                                          const detail::train_parameters<Task>& params,
+                                          const table& data,
+                                          const table& resp) {
+    using dal::detail::check_mul_overflow;
+
+    using model_t = model<Task>;
+    using model_impl_t = detail::model_impl<Task>;
+
+    auto opt_impl = detail::get_optimizer_impl(desc);
+
+    if (!opt_impl) {
+        throw internal_error{ dal::detail::error_messages::unknown_optimizer() };
+    }
+
+    auto queue = ctx.get_queue();
+
+    ONEDAL_PROFILER_TASK(log_reg_train_kernel, queue);
```

Comment:
I know this currently matches style of linreg but I'd suggest putting onedal profiler task definition at the top of the function (line 48) - it's a bit more intuitive that way